spi/common/http-spi/src/main/java/org/eclipse/edc/spi/http/FallbackFactory.java

```diff
@@ -0,0 +1,32 @@
+/*
+ *  Copyright (c) 2022 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.spi.http;
+
+import dev.failsafe.Fallback;
+import okhttp3.Request;
+import okhttp3.Response;
+
+/**
+ * Factory for {@link Fallback} instances given a {@link Request}
+ */
+public interface FallbackFactory {
```

Comment:
i could see this also being useful for other things than just HTTP requests, e.g. when we use some client SDK like Azure or AWS, were we typically get SDK exceptions or some status enum rather than HTTP status codes.  
May be something to look at in the future.